### PR TITLE
feat(AutoFocusZone): added simple AutoFocusZone component for focusing the first focusable element or the element with the provided selector

### PR DESCRIPTION
**Pull request by mnajdova**
Added AutoFocusZone component and popupAutoFocusBehavior.

### CODE REVIEW

File: packages/react/src/lib/accessibility/FocusZone/AutoFocusZone.tsx [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+import * as PropTypes from 'prop-types'
+import * as _ from 'lodash'
+
+import { getNextElement, focusAsync } from './focusUtilities'
+
+import { AutoFocusZoneProps } from './AutoFocusZone.types'
+import getUnhandledProps from '../../getUnhandledProps'
+import getElementType from '../../getElementType'
+import * as customPropTypes from '../../customPropTypes'
+import { callable } from '../../index'
```
Comment (layershifter) [inline packages/react/src/lib/accessibility/FocusZone/AutoFocusZone.tsx]:
```suggestion
import callable from '../../callable'
```

To simplify imports resolution

File: packages/react/src/lib/accessibility/FocusZone/AutoFocusZone.tsx [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    firstFocusableSelector: PropTypes.string,
+  }
+
+  public componentDidMount(): void {
+    this.findElementAndFocusAsync()
+  }
+
+  public render(): JSX.Element {
+    const unhandledProps = getUnhandledProps(
+      { handledProps: [..._.keys(AutoFocusZone.propTypes)] },
```
Comment (layershifter) [inline packages/react/src/lib/accessibility/FocusZone/AutoFocusZone.tsx]:
```suggestion
      { handledProps: AutoFocusZone.handledProps },
```

```jsx
static handledProps = _.keys(AutoFocusZone.propTypes)
```

File: packages/react/src/lib/accessibility/FocusZone/AutoFocusZone.tsx [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+import { callable } from '../../index'
+import Ref from '../../../components/Ref/Ref'
+
+/** AutoFocusZone is used to focus the first element inside it's children */
+export class AutoFocusZone extends React.Component<AutoFocusZoneProps, {}> {
+  private root = React.createRef<HTMLElement>()
+
+  static propTypes = {
+    as: customPropTypes.as,
+    firstFocusableSelector: PropTypes.string,
```
Comment (layershifter) [inline packages/react/src/lib/accessibility/FocusZone/AutoFocusZone.tsx]:
```suggestion
    firstFocusableSelector: PropTypes.oneOfType([PropTypes.func, PropTypes.string]),
```

File: packages/react/src/lib/accessibility/FocusZone/AutoFocusZone.tsx [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+import { getNextElement, focusAsync } from './focusUtilities'
+
+import { AutoFocusZoneProps } from './AutoFocusZone.types'
+import getUnhandledProps from '../../getUnhandledProps'
+import getElementType from '../../getElementType'
+import * as customPropTypes from '../../customPropTypes'
+import { callable } from '../../index'
+import Ref from '../../../components/Ref/Ref'
+
+/** AutoFocusZone is used to focus the first element inside it's children */
```
Comment (layershifter) [inline packages/react/src/lib/accessibility/FocusZone/AutoFocusZone.tsx]:
```suggestion
/** AutoFocusZone is used to focus the first element inside it's children. */
```

File: packages/react/test/specs/lib/AutoFocusZone-test.tsx [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+          >
+            <button className={'f'}>f</button>
+            <FocusZone>
+              <button className={'a'}>a</button>
+              <button className={'b'}>b</button>
+            </FocusZone>
+          </AutoFocusZone>
+          <button className={'z'}>z</button>
+        </div>,
+      ) as HTMLElement
```
Comment (layershifter) [inline packages/react/test/specs/lib/AutoFocusZone-test.tsx:73]: Can we use enzyme for tests?

Comment (mnajdova) [reply packages/react/test/specs/lib/AutoFocusZone-test.tsx:73]: I was copying and trying to keep these tests consistent with the other focus zone tests.. If we decide later to move all of them to enzyme we will do it for all of them.

File: packages/react/src/lib/accessibility/FocusZone/AutoFocusZone.tsx [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+import { getNextElement, focusAsync } from './focusUtilities'
+
+import { AutoFocusZoneProps } from './AutoFocusZone.types'
+import getUnhandledProps from '../../getUnhandledProps'
+import getElementType from '../../getElementType'
+import * as customPropTypes from '../../customPropTypes'
+import callable from '../../callable'
+import Ref from '../../../components/Ref/Ref'
+
+/** AutoFocusZone is used to focus the first element inside it's children. */
```
Comment (sophieH29) [inline packages/react/src/lib/accessibility/FocusZone/AutoFocusZone.tsx]: Not only first, but als by selector
...`is used to focus inner element on mount`...

File: packages/react/src/lib/accessibility/FocusZone/AutoFocusZone.types.tsx [hunk 7]
```diff
@@ -0,0 +1,13 @@
+import * as React from 'react'
+
+export interface AutoFocusZoneProps extends React.HTMLAttributes<HTMLDivElement> {
+  /**
+   * Element type the root element will use. Default is "div".
+   */
+  as?: React.ReactType
+
+  /**
+   * Indicates the selector for first focusable item.  Only applies if focusPreviouslyFocusedInnerElement == false.
```
Comment (sophieH29) [inline packages/react/src/lib/accessibility/FocusZone/AutoFocusZone.types.tsx]: No needd `Only applies if focusPreviouslyFocusedInnerElement == false.`

**Review by sophieH29 — APPROVED** [review]
Looks very good to me

File: packages/react/src/lib/accessibility/FocusZone/AutoFocusZone.types.tsx (hunk 7 above)
Comment (mnajdova) [reply packages/react/src/lib/accessibility/FocusZone/AutoFocusZone.types.tsx]: Changed to: `AutoFocusZone is used to focus inner element on mount (the first focusable or specified by the firstFocusableSelector prop).`

File: packages/react/src/lib/accessibility/FocusZone/AutoFocusZone.tsx (hunk 6 above)
Comment (mnajdova) [reply packages/react/src/lib/accessibility/FocusZone/AutoFocusZone.tsx]: Changed to: `AutoFocusZone is used to focus inner element on mount.` You are right, the props description will indicate the details.

**Review by layershifter — APPROVED** [review]

**Pull request merged**